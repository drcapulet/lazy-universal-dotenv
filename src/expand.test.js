import { getEnvironment } from "."

const snapshotOpts = {}

test("Supports expand of variable from .env", () => {
  const { raw, stringified, webpack } = getEnvironment({ nodeEnv: 'expansion1' })
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
})

test("Supports expand of variable from environment", () => {
  process.env.APP_DATA = "yellow"
  const { raw, stringified, webpack } = getEnvironment({ nodeEnv: 'expansion2' })
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
  delete process.env.APP_DATA
})
