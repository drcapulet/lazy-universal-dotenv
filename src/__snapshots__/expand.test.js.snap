// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supports expand of variable from .env 1`] = `
Object {
  "APP_TEST_OVERRIDE": "original",
  "APP_TEST_PATH": "one/two/three.js",
  "BINARY_PATH": "/usr/local/bin",
  "EXPAND_PATH": "/usr/bin:/usr/local/bin",
  "TEST_IGNORE": "123",
}
`;

exports[`Supports expand of variable from .env 2`] = `
Object {
  "APP_TEST_OVERRIDE": "\\"original\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "BINARY_PATH": "\\"/usr/local/bin\\"",
  "EXPAND_PATH": "\\"/usr/bin:/usr/local/bin\\"",
  "TEST_IGNORE": "\\"123\\"",
}
`;

exports[`Supports expand of variable from environment 1`] = `
Object {
  "APP_TEST_OVERRIDE": "original",
  "APP_TEST_PATH": "one/two/three.js",
  "EXPAND_DATA": "red,yellow",
  "TEST_IGNORE": "123",
}
`;

exports[`Supports expand of variable from environment 2`] = `
Object {
  "APP_TEST_OVERRIDE": "\\"original\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "EXPAND_DATA": "\\"red,yellow\\"",
  "TEST_IGNORE": "\\"123\\"",
}
`;
